reactive/containerd.py

```diff
@@ -185,6 +199,8 @@ def charm_status():
         status.blocked("Failed to fetch nvidia_apt_key_urls.")
     elif is_state("containerd.nvidia.missing_package_list"):
         status.blocked("No NVIDIA packages selected to install.")
+    elif _needs_gpu_reboot():
+        status.blocked("May need reboot to activate GPU.")
```

Comment:
this will be checked on every-exit.  Is that too often?  if so maybe it should be a flag

```containerd.nvidia.needs_reboot`` maybe?

Reply:
I think if users are updating packages using `apt` in the VM, it should be OK to check if a reboot is needed. On the other hand, if users are expected to use the action to upgrade the nvidia packages, it's better to use a flag.

Reply:
the package update can be triggered 2 ways:
* a config change on the application changes the package (`cuda-drivers-515` -> `cuda-drivers-520`)
* an upgrade action on the unit where there's an upgrade available (`cuda-drivers` points to something new)

I feel like calling `_needs_gpu_reboot` running on every hook and calling `nvidia-smi` on every hook COULD be overboard.  Likely, the drivers won't change that often


Perhaps a flag check would be better here... changing to "draft"

Reply:
In that case, a flag check sounds great!